src/Uuid/Uuid.php

```diff
@@ -0,0 +1,274 @@
+            // 32 bits for "time_low"
+            random_int(0, 0xffffffff),
+
+            // 16 bits for "time_mid"
+            random_int(0, 0xffff),
+
+            // 16 bits for "time_hi_and_version",
+            // four most significant bits holds version number 4
+            random_int(0, 0x0fff) | 0x4000,
+
+            // 16 bits:
+            // * 8 bits for "clk_seq_hi_res",
+            // * 8 bits for "clk_seq_low",
+            // two most significant bits holds zero and one for variant DCE1.1
+            random_int(0, 0x3fff) | 0x8000,
+
+            // 48 bits for "node"
+            random_int(0, 0xffffffffffff)
+        );
+    }
+
+    /**
+     * @see http://tools.ietf.org/html/rfc4122#section-4.2.2
+     */
+    private static function uuid_generate_time(): string
+    {
+        // https://tools.ietf.org/html/rfc4122#section-4.1.4
+        // 0x01b21dd213814000 is the number of 100-ns intervals between the
+        // UUID epoch 1582-10-15 00:00:00 and the Unix epoch 1970-01-01 00:00:00.
+        $offset = 0x01b21dd213814000;
+        $timeOfDay = gettimeofday();
+        $time = ($timeOfDay['sec'] * 10000000) + ($timeOfDay['usec'] * 10) + $offset;
+
+        // https://tools.ietf.org/html/rfc4122#section-4.1.5
+        // We are using a random data for the sake of simplicity: since we are
+        // not able to get a super precise timeOfDay as a unique sequence
+        $clockSeq = random_int(0, 0x3fff);
+
+        if (\function_exists('apcu_fetch')) {
+            $node = apcu_fetch('__symfony_uuid_node');
+            if (false === $node) {
+                $node = random_int(0, 0xffffffffffff);
+                if (false === apcu_store('__symfony_uuid_node', $node)) {
+                    goto static_node;
+                }
+            }
+        } else {
+            static_node:
+            static $node;
+            if (null === $node) {
+                $node = random_int(0, 0xffffffffffff);
+            }
+        }
+
+        return sprintf('%08x-%04x-%04x-%04x-%012x',
+            // 32 bits for "time_low"
+            $time & 0xffffffff,
+
+            // 16 bits for "time_mid"
+            ($time >> 32) & 0xffff,
+
+            // 16 bits for "time_hi_and_version",
+            // four most significant bits holds version number 1
+            ($time >> 48) | 1 << 12,
+
+            // 16 bits:
+            // * 8 bits for "clk_seq_hi_res",
+            // * 8 bits for "clk_seq_low",
+            // two most significant bits holds zero and one for variant DCE1.1
+            $clockSeq | 0x8000,
+
+            // 48 bits for "node"
+            $node
+        );
+    }
+
+    private static function uuid_parse_as_array(string $uuid)
+    {
+        if (36 != \strlen($uuid)) {
+            return null;
+        }
+
+        static $regex = '{^(?<time_low>[0-9a-f]{8})-(?<time_mid>[0-9a-f]{4})-(?<time_hi_and_version>[0-9a-f]{4})-(?<clock_seq>[0-9a-f]{4})-(?<node>[0-9a-f]{12})$}i';
```

Comment:
do we even need static variable, or should we inline the regex ?

Reply:
I did it for readability, but indeed :+1: I will move it

Reply:
Fixed